README.md

```diff
@@ -1,349 +1,422 @@
 
-Plays nice with [express-slow-down](https://www.npmjs.com/package/express-slow-down).
-
-Note: this module does not share state with other processes/servers by default. It also buckets all requests to an internal clock rather than starting a new timer for each end-user. It's fine for abuse-prevention but might not produce the desired effect when attempting to strictly enforce API rate-limits or similar. If you need a more robust solution, I recommend using an external store:
-
-### Stores
-
-- Memory Store _(default, built-in)_ - stores hits in-memory in the Node.js process. Does not share state with other servers or processes, and does not start a separate timer for each end user.
-- [Redis Store](https://npmjs.com/package/rate-limit-redis)
-- [Memcached Store](https://npmjs.org/package/rate-limit-memcached)
-- [Mongo Store](https://www.npmjs.com/package/rate-limit-mongo)
-- [Precise Memory Store](https://www.npmjs.com/package/precise-memory-rate-limit) - similar to the built-in memory store except that it stores a distinct timestamp for each IP rather than bucketing them together.
+> Note: this module does not share state with other processes/servers by
+> default. If you need a more robust solution, I recommend using an external
+> store. See the [`stores` section](#store) below for a list of external stores.
 
 ### Alternate Rate-limiters
 
-This module was designed to only handle the basics and didn't even support external stores initially. These other options all are excellent pieces of software and may be more appropriate for some situations:
+This module was designed to only handle the basics and didn't even support
+external stores initially. These other options all are excellent pieces of
+software and may be more appropriate for some situations:
 
 - [rate-limiter-flexible](https://www.npmjs.com/package/rate-limiter-flexible)
 - [express-brute](https://www.npmjs.com/package/express-brute)
 - [rate-limiter](https://www.npmjs.com/package/express-limiter)
 
 ## Install
 
+From the npm registry:
+
+```sh
+# Using npm
+> npm install express-rate-limit
+# Using yarn or pnpm
+> yarn/pnpm add express-rate-limit
+```
+
+From Github Releases:
+
 ```sh
-$ npm install --save express-rate-limit
+# Using npm
+> npm install https://github.com/nfriedly/express-rate-limit/releases/download/v{version}/express-rate-limit.tgz
+# Using yarn or pnpm
+> yarn/pnpm add https://github.com/nfriedly/express-rate-limit/releases/download/v{version}/express-rate-limit.tgz
 ```
 
+Replace `{version}` with the version of the package that you want to your, e.g.:
+`6.0.0`.
+
+> **This library (v6.0.0 or greater) is now pure ESM**. CommonJS packages will
+> not be able to synchronously import this package, and it is recommended to use
+> the package with NodeJS version 12 or greater. Take a look at this
+> [article](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c)
+> for more details.
```

Comment:
Isn't there a way to ship both via a small wrapper? So that users doing `const rateLimit = require("express-rate-limit");` still get the same behavior instead of having to change to `import` or `require(...).default` ?

Reply:
Or, if not a wapper, just run the type script compiler twice: once in CJS mode, and once in ESM mode, and then ship both outputs. 

Basically, I want the majority of existing users to be able to upgrade to the 6.0 release without having to make changes to their code.

Reply:
> Or, if not a wapper, just run the type script compiler twice: once in CJS mode, and once in ESM mode, and then ship both outputs.

Sure, we could do that. Maybe something like this in package.json:

```
"module": "dist/esm/index.js",
"main": "dist/cjs/index.js",
"exports": {
    ".": {
        "import": "./dist/esm/index.js",
        "require": "./dist/cjs/index.js"
    }
},
```

> Basically, I want the majority of existing users to be able to upgrade to the 6.0 release without having to make changes to their code.

👍🏾